phases/13-tools-and-protocols/07-building-an-mcp-server/code/tests/test_main.py

```
@@ -0,0 +1,127 @@
+            with self.subTest(request_id=request_id):
+                message = main.make_request(2, "tools/list")
+                message["id"] = request_id
+                response = main.dispatch(message)
+                self.assertEqual(response["id"], None)
+                self.assertEqual(response["error"]["code"], -32600)
+
+    def test_typescript_ids_are_limited_to_safe_integer_bounds(self) -> None:
+        node = shutil.which("node")
+        if node is None:
+            self.skipTest("Node.js is unavailable")
+        version = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)